src/states/common/unapproved.rs

```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use super::Bootstrapped;
+use crate::{
+    error::RoutingError,
+    id::PublicId,
+    messages::{HopMessage, RoutingMessage},
+    routing_message_filter::FilteringResult,
+};
+
+pub trait Unapproved: Bootstrapped {
+    fn filter_hop_message(
+        &mut self,
+        hop_msg: HopMessage,
+        pub_id: PublicId,
+    ) -> Result<Option<RoutingMessage>, RoutingError> {
+        hop_msg.verify(pub_id.signing_public_key())?;
+
+        let signed_msg = hop_msg.content;
+        signed_msg.check_integrity(self.min_section_size())?;
+
+        let routing_msg = signed_msg.into_routing_message();
+        let in_authority = self.in_authority(&routing_msg.dst);
+        if in_authority {
+            self.send_ack(&routing_msg, 0);
```

Comment:
This was not in src/states/client.rs.
Only in src/states/joining_node.rs, could you explain the change in behaviour

Reply:
There is no change, the difference is handled with the `send_ack` param now.

Reply:
Sorry I do not understand what you mean, can you be more specific.

Reply:
Sorry, sure: if you look at the code before this PR, then you can see that the `handle_hop_message` method in [Client](https://github.com/maidsafe/routing/blob/master/src/states/client.rs#L207-L241) and in [JoiningNode](https://github.com/maidsafe/routing/blob/master/src/states/joining_node.rs#L219-L255) are almost identical. The only difference is that `JoiningNode` does `send_ack` but `Client` doesn't. So I extracted that logic into `filter_hop_message` where the `send_ack` difference is handled using additional argument.

Reply:
I was looking that the commit https://github.com/maidsafe/routing/pull/1613/commits/340c8328a7b07e001cc5a5b2171319ed03b02790 which did the extraction but not have it, I guess you fixed it in a latter commit then.